core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1225,7 +1225,7 @@ private int acquireSubsetBatchRecords(
                     break;
                 }
 
-                if (offsetState.getValue().state != RecordState.AVAILABLE) {
+                if (offsetState.getValue().state != RecordState.AVAILABLE || offsetState.getValue().rollbackState() != null) {
                     log.trace("The offset is not available skipping, offset: {} batch: {}"
```

Comment:
I would change the log line here for commonality with the others in terms of the order of inserts. I suggest:
```
log.trace("The offset {} is not available in share partition: {}-{}, skipping: {}",
    offsetState.getKey(), groupId, topicIdPartition, inFlightBatch);
```